backend/middleware/inputValidation.ts

```
@@ -0,0 +1,256 @@
+  if (!filename || typeof filename !== 'string') {
+    throw new Error('Invalid filename');
+  }
+  
+  // SECURITY: Remove all path separators and parent directory references
+  return filename
+    .replace(/\.\./g, '') // Remove ..
+    .replace(/[/\\]/g, '') // Remove / and \
+    .replace(/^\.+/, '') // Remove leading dots
+    // eslint-disable-next-line no-control-regex
+    .replace(/[<>:"|?*\x00-\x1f]/g, '') // Remove invalid filename chars
+    .trim();
```

Biome (2.1.2)

[error] Unexpected control character in a regular expression.

Control characters are unusual and potentially incorrect inputs, so they are disallowed.

(lint/suspicious/noControlCharactersInRegex)